specs/phase0/fork-choice.md

```diff
@@ -284,12 +280,8 @@ def on_block(store: Store, signed_block: SignedBeaconBlock) -> None:
     # Add new block to the store
     store.blocks[hash_tree_root(block)] = block
     # Check block is a descendant of the finalized block
-    assert (
-        get_ancestor(store, hash_tree_root(block), store.blocks[store.finalized_checkpoint.root].slot) ==
-        store.finalized_checkpoint.root
-    )
-    # Check that block is later than the finalized epoch slot
-    assert block.slot > compute_start_slot_at_epoch(store.finalized_checkpoint.epoch)
```

Comment:
Is it worth keeping this line, as a way to reduce possible calls to `get_ancestor`, which is a more expensive operation?

Reply:
added and noted it is an optimization